src/Featurizers/LagLeadOperatorFeaturizer.h

```diff
@@ -0,0 +1,412 @@
+        Components::FilterDecoratorEstimator<std::tuple<std::vector<std::string> const &, Microsoft::Featurizer::RowMajMatrix<typename Microsoft::Featurizer::Traits<InputT>::nullable_type>>, 1>
+    > {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType = 
+        Components::PipelineExecutionEstimatorImpl<
+            Components::GrainEstimatorImpl<std::vector<std::string>, LagLeadOperatorEstimator<InputT, MaxNumTrainingItemsV>>,
+            Components::FilterDecoratorEstimator<std::tuple<std::vector<std::string> const &, Microsoft::Featurizer::RowMajMatrix<typename Microsoft::Featurizer::Traits<InputT>::nullable_type>>, 1>
+        >;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    GrainedLagLeadOperatorEstimator(AnnotationMapsPtr pAllColumnAnnotations, std::uint32_t horizon, std::vector<std::int64_t> offsets);
+    ~GrainedLagLeadOperatorEstimator(void) override = default;
+};
+
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// |
+// |  Implementation
+// |
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+
+// ----------------------------------------------------------------------
+// |
+// |  LagLeadOperatorTransformer
+// |
+// ----------------------------------------------------------------------
+template <typename T>
+LagLeadOperatorTransformer<T>::LagLeadOperatorTransformer(std::uint32_t horizon, std::vector<std::int64_t> offsets) :
+    _horizon(
+        std::move(
+            [&horizon](void) -> std::uint32_t & {
+                if (horizon == 0) {
+                    throw std::invalid_argument("Horizon cannot be 0!");
+                }
+                return horizon;
+            }()
+    )),
+    _offset_delta(
+        std::move(
+            [&offsets](void) -> std::int64_t {
+                if (offsets.size() == 0) {
+                    throw std::invalid_argument("Offsets is empty!");
+                }
+                return *std::min_element(offsets.cbegin(), offsets.cend()) < 0 ? *std::min_element(offsets.cbegin(), offsets.cend()) : 0;
+            }()
+        )
+    ),
+    // calibrate offsets to minimum offset = 0
+    _offsets(
+        std::move(
+            [&offsets](void) -> std::vector<std::int64_t> & {
```

Comment:
>offsets [](start = 14, length = 7)

Error if offsets is empty.

Reply:
If offsets is empty, it will be caught in line 226.